### PR TITLE
Notification type for recovery

### PR DESCRIPTION
**Pull request by james-toussaint**

### CODE REVIEW

File: src/main/java/com/iexec/common/chain/IexecHubAbstractService.java [hunk 1]
```diff
@@ -315,4 +319,35 @@ protected boolean isStatusValidOnChainAfterPendingReceipt(String chainTaskId, Ch
         return false;
     }
 
+    public Optional<TaskDescription> getTaskDescriptionFromChain(String chainTaskId) {
+
+        Optional<ChainTask> optionalChainTask = getChainTask(chainTaskId);
+        if (!optionalChainTask.isPresent()) {
+            log.info("Failed to retrieve AvailableReplicate, ChainTask error  [chainTaskId:{}]", chainTaskId);
```
Comment (zguesmi) [inline src/main/java/com/iexec/common/chain/IexecHubAbstractService.java]: maybe we replace "AvailableReplicate" with "task description" in the logs for consistency sake.

File: src/main/java/com/iexec/common/chain/IexecHubAbstractService.java [hunk 2]
```diff
@@ -315,4 +319,35 @@ protected boolean isStatusValidOnChainAfterPendingReceipt(String chainTaskId, Ch
...
+            return Optional.empty();
+        }
+
+        ChainTask chainTask = optionalChainTask.get();
+
+        Optional<ChainDeal> optionalChainDeal = getChainDeal(chainTask.getDealid());
+        if (!optionalChainDeal.isPresent()) {
+            log.info("Failed to retrieve AvailableReplicate, ChainDeal error  [chainTaskId:{}]", chainTaskId);
```
Comment (zguesmi) [inline src/main/java/com/iexec/common/chain/IexecHubAbstractService.java]: Same here

File: src/main/java/com/iexec/common/utils/MultiAddressHelper.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    private MultiAddressHelper() {
+        throw new UnsupportedOperationException();
+    }
+
+    public static String convertToURI(String hexaString) {
+        // Currently, only the IPFS format is supported in the multiaddress field in the smart contract, so
+        // if the MultiAddress object can't be constructed, it means it is an HTTP address.
+        try {
+            MultiAddress address = new MultiAddress(Numeric.hexStringToByteArray(hexaString));
+            return IPFS_GATEWAY_URI + address.toString();
```
Comment (zguesmi) [inline src/main/java/com/iexec/common/utils/MultiAddressHelper.java:19]: Does it need a "slash" here in the middle?

File: src/main/java/com/iexec/common/chain/IexecHubAbstractService.java (hunk 1 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/chain/IexecHubAbstractService.java]: Thanks!

File: src/main/java/com/iexec/common/chain/IexecHubAbstractService.java (hunk 2 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/chain/IexecHubAbstractService.java]: Thanks2!

File: src/main/java/com/iexec/common/utils/MultiAddressHelper.java (hunk 3 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/utils/MultiAddressHelper.java:19]: Just moved some code from Worker to Common, nothing changed (https://github.com/iExecBlockchainComputing/iexec-worker/blob/c5cf832c55f8eceb1daa90fe46f020741d315448/src/main/java/com/iexec/worker/utils/MultiAddressHelper.java#L20)

Comment (zguesmi) [reply src/main/java/com/iexec/common/utils/MultiAddressHelper.java:19]: ah okey!

**Review by zguesmi — APPROVED** [review]

**Pull request merged**